### PR TITLE
do not error if iam role already exists during setup

### PR DESCRIPTION
**Pull request by Vaishaal**
fixing  #236

### CODE REVIEW

File: pywren/scripts/pywrencli.py [hunk 1]
```diff
@@ -140,20 +140,24 @@ def create_role(ctx):
     config_filename = ctx.obj['config_filename']
     config = pywren.wrenconfig.load(config_filename)
 
-    iamclient = boto3.resource('iam')
+    iam= boto3.resource('iam')
```
Comment (shivaram) [inline pywren/scripts/pywrencli.py]: space after `iam`

File: pywren/scripts/pywrencli.py [hunk 2]
```diff
@@ -140,20 +140,24 @@ def create_role(ctx):
...
+    iamclient = boto3.client('iam')
     json_policy = json.dumps(pywren.wrenconfig.basic_role_policy)
     role_name = config['account']['aws_lambda_role']
-    iamclient.create_role(RoleName=role_name,
-                          AssumeRolePolicyDocument=json_policy)
-    more_json_policy = json.dumps(pywren.wrenconfig.more_permissions_policy)
+    roles = [x for x in iamclient.list_roles()["Roles"] if x["RoleName"] == role_name]
+    if (len(roles) == 0):
+        iam.create_role(RoleName=role_name,
+                              AssumeRolePolicyDocument=json_policy)
```
Comment (shivaram) [inline pywren/scripts/pywrencli.py]: align `AssumeRolePolicy` with `RoleName` ?

File: pywren/scripts/pywrencli.py [hunk 3]
```diff
@@ -140,20 +140,24 @@ def create_role(ctx):
...
+        AWS_ACCOUNT_ID = config['account']['aws_account_id']
+        AWS_REGION = config['account']['aws_region']
+        more_json_policy = more_json_policy.replace("AWS_ACCOUNT_ID", str(AWS_ACCOUNT_ID))
+        more_json_policy = more_json_policy.replace("AWS_REGION", AWS_REGION)
+
+        iam.RolePolicy(role_name, '{}-more-permissions'.format(role_name)).put(
+            PolicyDocument=more_json_policy)
 
-    iamclient.RolePolicy(role_name, '{}-more-permissions'.format(role_name)).put(
-        PolicyDocument=more_json_policy)
```
Comment (shivaram) [inline pywren/scripts/pywrencli.py]: print a line saying something like `using existing IAM role ` or something like that ?

**Comment by shivaram** [comment]
Tests are failing with `botocore.exceptions.ClientError: An error occurred (AccessDenied) when calling the ListRoles operation: User: arn:aws:iam::783175685819:user/travis_pywren_tester is not authorized to perform: iam:ListRoles on resource: arn:aws:iam::783175685819:role/`

**Comment by Vaishaal** [comment]
@ericmjonas please give travis the appropriate permissions

**Comment by ericmjonas** [comment]
@Vaishaal Vaishaal I added the ListRoles access to the IAM section of the travis test user so now it's
```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "VisualEditor0",
            "Effect": "Allow",
            "Action": [
                "iam:DeleteInstanceProfile",
                "iam:PassRole",
                "iam:GetInstanceProfile",
                "iam:DeleteRolePolicy",
                "iam:RemoveRoleFromInstanceProfile",
                "iam:ListRoles",
                "iam:CreateRole",
                "iam:DeleteRole",
                "iam:PutRolePolicy",
                "iam:GetUser",
                "iam:AddRoleToInstanceProfile"
            ],
            "Resource": "*"
        }
    ]
}
```

let me know if you need somehting else

**Pull request merged**